src/7-to-8/major-changes/compatibility-mode.rst

```diff
@@ -39,22 +40,101 @@ The old ``suite.rc`` filename triggers a backward compatibility mode in which:
     :ref:`*optional* <User Guide Optional Outputs>` by the new ``?`` syntax)
 
 
+.. _compat_required_changes:
+
 Required Changes
 ----------------
 
 Providing your Cylc 7 workflow does not use syntax that was deprecated at Cylc 7,
-you may be able to run it with Cylc 8 without any modifications to begin with.
+you may be able to run it using Cylc 8 without any modifications while in
+compatibility mode.
+
+First, run ``cylc validate`` **with Cylc 7** on your ``suite.rc`` workflow
+to check for deprecation warnings and fix those before validating with Cylc 8.
+See :ref:`below <compat.eg.c7val>` for an example.
+
+.. warning::
+
+   ``cylc validate`` operates on the processed ``suite.rc``, which
+   means it will not detect any deprecated syntax that is inside a
+   currently-unused Jinja2/EmPy ``if...else`` branch.
+
+Some workflows may require modifications to either upgrade to Cylc 8 or make
+interoperable with Cylc 8 backward compatibility mode. Read on for more details.
+
+
+Cylc commands in task scripts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Check for any use of Cylc commands in task scripting. Some Cylc 7 commands
+have been removed and some others now behave differently.
+However, ``cylc message`` and ``cylc broadcast`` have *not* changed.
+See the :ref:`full list of command line interface changes<MajorChangesCLI>`
+and see :ref:`below <compat.eg.cylc-commands>` for an example.
+
+
+Python 2 to 3
+^^^^^^^^^^^^^
+
+Whereas Cylc 7 runs using Python 2, Cylc 8 runs using Python 3. This affects:
+- modules imported in Jinja2
+- Jinja2 filters, tests and globals
+- custom xtrigger functions
+
+Note that task scripts are not affected - they can run using an independent
```

Comment:
```suggestion
Note that task scripts are not affected - they run in an independent
```
(Not necessarily 100% true for local background jobs, but users should think of it like this anyway).